File: backend/api/src/main/kotlin/ru/sigma/api/GameApi.kt
package ru.sigma.api

import ru.sigma.common.model.Coordinate
import ru.sigma.game.domain.dto.AfterShotStateDto
import ru.sigma.game.domain.dto.GameDto

interface GameApi {
    fun getGame(id: Long): GameDto
    fun makeShot(id: Long, coords: Coordinate): AfterShotStateDto
    fun leaveGame(id: Long)
}

File: frontend/sea-battle/app/features/game/api/gameApi.ts
import {httpClient, httpClientToken} from "~/shared/api/client";
import {API_ENDPOINTS} from "~/shared/api/endpoints";
import type { GameDto, Coordinate } from '~/processes/game/model/types';

export const gameApi = {
    getGameState: (id:string) =>
        httpClientToken.get<GameDto>(API_ENDPOINTS.GAME.GET_GAME_STATE(id), {
            headers: {
                'Content-Type': 'application/json',
            }
        }),

    leaveGame: (id:string) =>
        httpClientToken.post<GameDto>(API_ENDPOINTS.GAME.LEAVE_GAME(id), {
            headers: {
                'Content-Type': 'application/json',
            }
        }),

    shot: (id:string, coords: Coordinate) =>
        httpClientToken.post<GameDto>(API_ENDPOINTS.GAME.SHOT(id), coords, {
            headers: {
                'Content-Type': 'application/json',
            }
        }),
};